ecs-cli/modules/cli/image/image_app_test.go

```diff
@@ -429,6 +429,46 @@ func TestSplitImageNameWithURI(t *testing.T) {
 	assert.NoError(t, err, "Error splitting image name")
 }
 
+func TestSplitImageNameWithFIPSEndpoint(t *testing.T) {
```

Comment:
Since these are all roughly the same logic, I'd recommend changing them to a single table-driven test:

```go
func TestSplitImageName(t *testing.T) {
	tests := []struct{
		name       string
		uri        string
		repository string
		tag        string
	}{{
		name: "FIPS endpoint",
		uri: "012345678912.dkr.ecr-fips.us-gov-west-1.amazonaws.com",
		repository: repository,
	},{
		name: "FIPS endpoint and slash in image name",
		uri: "012345678912.dkr.ecr-fips.us-gov-west-1.amazonaws.com"
		repository: repositoryWithSlash,
	}}

	for _, tc := range tests {
		t.Run(tc.name, func(t *testing.T) {
			expectedImage := uri + "/" + repository
			if tag != "" {
				expectedImage += ":" + tag
			}
			observedRegistryURI, observedRepo, observedTag, err := splitImageName(expectedImage, "[:|@]", "format")
			assert.Equal(t, uri, observedRegistryURI, "RegistryURI should match")
			assert.Equal(t, repository, observedRepo, "Repository should match)
			assert.Equal(t, tag, observedTag, "Tag should match")
			assert.NoError(t, err, "Error splitting image name")
		})
	}
}
```